### PR TITLE
House prices

### PR DESCRIPTION
**Pull request by LostFan123**
**Kaggle competition**
https://www.kaggle.com/c/house-prices-advanced-regression-techniques

**Goal**
Predict the sales price for each house. For each Id in the test set, predict the value of the SalePrice variable. 

**Submission File Format**
The file should contain a header and have the following format:
```
Id,SalePrice
1461,169000.1
1462,187724.1233
1463,175221
etc.
```

**Useful kernels**
[Comprehensive data exploration with Python](https://www.kaggle.com/pmarcelino/comprehensive-data-exploration-with-python)
[Regularized Linear Models](https://www.kaggle.com/apapiu/regularized-linear-models)
[A study on Regression applied to the Ames dataset](https://www.kaggle.com/juliencs/a-study-on-regression-applied-to-the-ames-dataset)
[House Prices EDA](https://www.kaggle.com/dgawlik/house-prices-eda)

**TODO:**
- [ ] find out what dummy variables for categorical features are
- [ ] keep categorical and quantitative columns separated
- [ ] check how values are distributed
- [ ] use seaborn for visualization 
- [ ] plot correlation matrix
- [ ] read on Spearman correlation
- [ ] try scatter plots (see 1st link)
- [ ] analyze missing data
- [ ] analyze outliers
- [ ] learn about skewness and kurtosis
- [ ] mind the normality 
- [ ] try Shapiro-Wilk test
- [ ] learn about Johnson's distribution
- [ ] learn about Box-Cox transformation
- [ ] learn about Fisher transformation
- [ ] learn about homoscedasticity
- [ ] use cross-validation
- [ ] try lasso model 
- [ ] try ridge model
- [ ] try ElasticNet
- [ ] try ANOVA
- [ ] detect most important coefficients (see 2nd link)
- [ ] try [xgboost](https://github.com/dmlc/xgboost)
- [ ] try neural networks with [Keras](https://keras.io/)
- [ ] find out if it's ok to add polynomial features for existing ones (e.g.: x - old; x², x³ - new )

### CODE REVIEW

**Comment by LostFan123** [comment]
**Dummy variable** - variable that takes the value 0 or 1 to indicate the absence or presence of some categorical effect that may be expected to shift the outcome.
[Wikipedia - Dummy variable](https://en.wikipedia.org/wiki/Dummy_variable_(statistics))
[Dummy variable trap in regression models](http://www.algosome.com/articles/dummy-variable-trap-regression.html)
[Simple Methods to deal with Categorical Variables in Predictive Modeling](https://www.analyticsvidhya.com/blog/2015/11/easy-methods-deal-categorical-variables-predictive-modeling/)
[The Use of Dummy Variables in Regression Analysis](https://www.moresteam.com/whitepapers/download/dummy-variables.pdf)

**Comment by LostFan123** [comment]
**Separating categorical data from quantitative in pandas**
`DataFrame.select_dtypes(include=None, exclude=None)`
_include, exclude_ : A list of dtypes or strings to be included/excluded. You must pass in a non-empty sequence for at least one of these.

- To select all numeric types use the numpy dtype numpy.number
- To select strings you must use the object dtype, but note that this will return all object dtype columns
- See the [numpy](https://docs.scipy.org/doc/numpy/reference/arrays.scalars.html) dtype hierarchy
- To select datetimes, use np.datetime64, ‘datetime’ or ‘datetime64’
- To select timedeltas, use np.timedelta64, ‘timedelta’ or ‘timedelta64’
- To select Pandas categorical dtypes, use ‘category’

`>>> df.select_dtypes(include=['float64'])`
